tests/test_particle_sets.py

```diff
@@ -93,34 +94,42 @@ def IncrLon(particle, fieldset, time, dt):
     assert np.allclose([p.lon for p in pset], np.arange(npart, 0, -1))
 
 
+@pytest.mark.parametrize('type', ['releasedt', 'timearr'])
 @pytest.mark.parametrize('mode', ['scipy', 'jit'])
 @pytest.mark.parametrize('repeatdt', range(1, 3))
 @pytest.mark.parametrize('dt', [-1, 1])
 @pytest.mark.parametrize('maxvar', [2, 4, 10])
-def test_pset_repeated_release_delayed_adding_deleting(fieldset, mode, repeatdt, tmpdir, dt, maxvar, runtime=10):
+def test_pset_repeated_release_delayed_adding_deleting(type, fieldset, mode, repeatdt, tmpdir, dt, maxvar, runtime=10):
     fieldset.maxvar = maxvar
 
     class MyParticle(ptype[mode]):
         sample_var = Variable('sample_var', initial=0.)
-    pset = ParticleSet(fieldset, lon=[0], lat=[0], pclass=MyParticle, repeatdt=repeatdt)
-    outfilepath = tmpdir.join("pfile_repeatdt")
-    pfile = pset.ParticleFile(outfilepath, outputdt=abs(dt))
+    if type == 'releasedt':
+        pset = ParticleSet(fieldset, lon=[0], lat=[0], pclass=MyParticle, repeatdt=repeatdt)
+    elif type == 'timearr':
+        pset = ParticleSet(fieldset, lon=np.zeros(runtime), lat=np.zeros(runtime), pclass=MyParticle, time=list(range(runtime)))
+    outfilepath = tmpdir.join("pfile_repeated_release")
+    pfile = pset.ParticleFile(outfilepath, outputdt=abs(dt), chunksizes=[len(pset), 1])
 
     def IncrLon(particle, fieldset, time, dt):
         particle.sample_var += 1.
         if particle.sample_var > fieldset.maxvar:
             particle.delete()
     for i in range(runtime):
         pset.execute(IncrLon, dt=dt, runtime=1., output_file=pfile)
-    assert np.allclose([p.sample_var for p in pset], np.arange(maxvar, -1, -repeatdt))
     ncfile = Dataset(outfilepath+".nc", 'r', 'NETCDF4')
     samplevar = ncfile.variables['sample_var'][:]
-    assert samplevar.shape == (runtime // repeatdt+1, min(maxvar+1, runtime)+1)
-    if repeatdt == 0:
-        # test whether samplevar[i, i+k] = k for k=range(maxvar)
-        for k in range(maxvar):
-            for i in range(runtime-k):
-                assert(samplevar[i, i+k] == k)
+    ncfile.close()
+    if type == 'releasedt':
+        assert samplevar.shape == (runtime // repeatdt+1, min(maxvar+1, runtime)+1)
+        assert np.allclose([p.sample_var for p in pset], np.arange(maxvar, -1, -repeatdt))
```

Comment:
Is it normal that we check p.sample_var for p in pset for `type == 'releasedt'` but not `type == 'timearr'` ?